presto-main/src/main/java/com/facebook/presto/tdigest/TDigest.java

```diff
@@ -0,0 +1,664 @@
+        }
+
+        checkArgument(weight[n - 1] > 1, format("Expected weight[n - 1] > 1, but was %s", weight[n - 1]));
+        checkArgument(index <= totalWeight, format("Expected index <= totalWeight, but index:%s > totalWeight:%s", index, totalWeight));
+        checkArgument(index >= totalWeight - weight[n - 1] / 2, format("Expected index >= totalWeight - weight[n - 1] / 2, but" +
+                "index:%s < %s", index, totalWeight - weight[n - 1] / 2));
+
+        // weightSoFar = totalWeight - weight[n-1]/2 (very nearly)
+        // so we interpolate out to max value ever seen
+        double z1 = index - totalWeight - weight[n - 1] / 2.0;
+        double z2 = weight[n - 1] / 2 - z1;
+        return weightedAverage(mean[n - 1], z1, max, z2);
+    }
+
+    public int centroidCount()
+    {
+        mergeNewValues();
+        return activeCentroids;
+    }
+
+    public Collection<Centroid> centroids()
+    {
+        compress();
+        return new AbstractCollection<Centroid>()
+        {
+            @Override
+            public Iterator<Centroid> iterator()
+            {
+                return new Iterator<Centroid>()
+                {
+                    int i;
+
+                    @Override
+                    public boolean hasNext()
+                    {
+                        return i < activeCentroids;
+                    }
+
+                    @Override
+                    public Centroid next()
+                    {
+                        Centroid rc = new Centroid(mean[i], (int) weight[i]);
+                        i++;
+                        return rc;
+                    }
+
+                    @Override
+                    public void remove()
+                    {
+                        throw new UnsupportedOperationException("Default operation");
+                    }
+                };
+            }
+
+            @Override
+            public int size()
+            {
+                return activeCentroids;
+            }
+        };
+    }
+
+    public double getCompressionFactor()
+    {
+        return publicCompression;
+    }
+
+    public long estimatedSerializedSizeInBytes()
+    {
+        compress();
+        return SIZE_OF_BYTE                             // format
+                + SIZE_OF_BYTE                          // type (e.g double, float, bigint)
+                + SIZE_OF_DOUBLE                        // min
+                + SIZE_OF_DOUBLE                        // max
+                + SIZE_OF_DOUBLE                        // compression factor
+                + SIZE_OF_DOUBLE                        // total weight
+                + SIZE_OF_INT                           // number of centroids
+                + SIZE_OF_DOUBLE * activeCentroids      // weight[], containing weight of each centroid
+                + SIZE_OF_DOUBLE * activeCentroids;     // mean[], containing mean of each centroid
+    }
+
+    public long estimatedInMemorySizeInBytes()
```

Comment:
type: there is an extra `In`

Reply:
That was done on purpose, consistent with QuantileDigest.java